classes/class.memberorder.php

```diff
@@ -1103,11 +1098,26 @@ function get_test_order() {
 		 * @return bool True if a subscription was created, false if not.
 		 */
 		private function create_subscription_for_order() {
+			global $pmpro_level;
+
+			// Make sure that this order is a part of a subscription.
 			if ( empty( $this->subscription_transaction_id ) ) {
 				// No subscription transaction ID, so we don't need to create a subscription.
 				return false;
 			}
 
+			// Make sure that this order has been completed.
+			if ( 'success' !== $this->status ) {
+				// The order is not complete, so we shouldn't create a subscription yet.
+				return false;
+			}
+
+			// Make sure that the user has the membership level for this order.
```

Comment:
During checkout, does the order get processed and membership get set up before the subscription is supposed to be set up? This logic might not be right, but I'd like to look at it closer with you.

Reply:
In the checkout preheader, the subscription is set up on 446:
https://github.com/strangerstudios/paid-memberships-pro/blob/5ca74f946d62c7783bb248fd8d128289b39f6d68/preheaders/checkout.php#L446

And then the membership level is changed at 664 and, if successful, the order is saved on 680:
https://github.com/strangerstudios/paid-memberships-pro/blob/5ca74f946d62c7783bb248fd8d128289b39f6d68/preheaders/checkout.php#L664-L681

So by the time that the order is saved, the initial payment should be processed, a subscription set up, and a membership assigned.